### ISSUE TITLE
Met unsupported operator of type Cast when quantize model get from quantize aware training

### ISSUE THREAD

**Issue by SystemErrorWang** [tensorflow/tensorflow#26478]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**

- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
yes, i tried to use quantize aware training, post training quantization and tflite converter to train and convert a custom mobilenetv2 model
- OS Platform and Distribution :
macos 10.13.6
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
None
- TensorFlow installed from (source or binary):
conda install tensorflow
- TensorFlow version (use command below):
tensorflow 1.13
- Python version:
python 3.6.8
- Bazel version (if compiling from source):
none
- GCC/Compiler version (if compiling from source):
none
- CUDA/cuDNN version:
none
- GPU model and memory:
none

i was training a mobilenetv2 model, using quantize aware training defined in tf.contrib.quantize, following the direction in [https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/quantize](url)

 tflite convert was used to convert the quantize aware trained model to 8bit fixed point. everything was fine until i saved frozen graph. i then use toco command line tool, and got this error massage:
"2019-03-08 16:13:24.208096: F ./tensorflow/lite/toco/toco_tooling.h:38] Check failed: s.ok() Unimplemented: this graph contains an operator of type Cast for which the quantized form is not yet implemented. Sorry, and patches welcome (that's a relatively fun patch to write, mostly providing the actual quantized arithmetic code for this op).
Fatal Python error: Aborted"

i also tried tflite converter python api, convert from session, convert from frozen graph and convert from saved models, and all got the same error message. by printing out ops in the gragh, i found some "Cast" ops in the graph, in batch norm layers mostly. i guess these cast operators were used building training graphs and folding batch norms, and should be deleted when building eval graph or freezing the graph. but i still get them in my frozen graph, so i would like to ask for some help to get rid of them (and other unsupported ops) and convert the model to 8bit

the nodes in graph_def is saved here [https://www.dropbox.com/s/mev825dzrzh7hrv/log.txt?dl=0](url) 
saved frozen graph is available here [https://www.dropbox.com/s/8xbropujwwxm5tb/frozen_graph.pb?dl=0](url)

**Comment by ai1361720220000** [comment]
hello, have you solved this problem of cast op?

**Comment by liyunlu0618** [comment]
> hello, have you solved this problem of cast op?

The contrib.quantize is no longer maintained. Please try the new keras based quantization aware training [API](https://www.tensorflow.org/model_optimization/guide/quantization/training_example) and let us know if you run into any issues. Thanks!

**Comment by tensorflowbutler** [comment]
Hi There,

 We are checking to see if you still need help on this issue, as you are using an older version of tensorflow(1.x) which is officially considered as end of life. We recommend that you upgrade to 2.4 or later version and let us know if the issue still persists in newer versions. 

 This issue will be closed automatically 7 days from now. If you still need help with this issue, Please open a new issue for any help you need against 2.x, and we will get you the right help.
